src/equations/polytropic_euler_2d.jl

```diff
@@ -0,0 +1,315 @@
+    rho_rr, v1_rr, v2_rr = cons2prim(u_rr, equations)
+    p_ll = equations.kappa * rho_ll^equations.gamma
+    p_rr = equations.kappa * rho_rr^equations.gamma
+
+    v_normal_ll = v1_ll * normal_direction[1] + v2_ll * normal_direction[2]
+    v_normal_rr = v1_rr * normal_direction[1] + v2_rr * normal_direction[2]
+
+    norm_ = norm(normal_direction)
+    # The v_normals are already scaled by the norm
+    lambda_min = v_normal_ll - sqrt(equations.gamma * p_ll / rho_ll) * norm_
+    lambda_max = v_normal_rr + sqrt(equations.gamma * p_rr / rho_rr) * norm_
+
+    return lambda_min, lambda_max
+end
+
+@inline function max_abs_speeds(u, equations::PolytropicEulerEquations2D)
+    rho, v1, v2 = cons2prim(u, equations)
+    c = sqrt(equations.gamma * equations.kappa * rho^(equations.gamma - 1))
+
+    return abs(v1) + c, abs(v2) + c
+end
+
+# Convert conservative variables to primitive
+@inline function cons2prim(u, equations::PolytropicEulerEquations2D)
+    rho, rho_v1, rho_v2 = u
+
+    v1 = rho_v1 / rho
+    v2 = rho_v2 / rho
+
+    return SVector(rho, v1, v2)
+end
+
+# Convert conservative variables to entropy
+@inline function cons2entropy(u, equations::PolytropicEulerEquations2D)
+    rho, rho_v1, rho_v2 = u
+
+    v1 = rho_v1 / rho
+    v2 = rho_v2 / rho
+    v_square = v1^2 + v2^2
+    p = pressure(u, equations)
+    # Form of the internal energy depends on gas type
+    if equations.gamma == 1.0 # isothermal gas
+        internal_energy = equations.kappa * log(rho)
+    else # equations.gamma > 1 # polytropic gas
+        internal_energy = equations.kappa * rho^equations.gamma /
+                          (equations.gamma - 1.0)
+    end
+
+    w1 = internal_energy + p / rho - 0.5 * v_square
+    w2 = v1
+    w3 = v2
+
+    return SVector(w1, w2, w3)
+end
+
+# Convert primitive to conservative variables
+@inline function prim2cons(prim, equations::PolytropicEulerEquations2D)
+    rho, v1, v2 = prim
+    rho_v1 = rho * v1
+    rho_v2 = rho * v2
+    return SVector(rho, rho_v1, rho_v2)
+end
+
+@inline function density(u, equations::PolytropicEulerEquations2D)
+    rho = u[1]
+    return rho
+end
+
+@inline function pressure(u, equations::PolytropicEulerEquations2D)
+    rho, rho_v1, rho_v2 = u
+    p = equations.kappa * rho^equations.gamma
+    return p
+end
+
+"""
+initial_condition_convergence_test(x, t, equations::PolytropicEulerEquations2D)
+
+Manufactured smooth initial condition used for convergence tests.
+"""
+function initial_condition_convergence_test(x, t, equations::PolytropicEulerEquations2D)
+    # manufactured initial condition from Winters (2019) [0.1007/s10543-019-00789-w]
+    # domain must be set to [0, 1] x [0, 1]
+    h = 8 + cos(2 * pi * x[1]) * sin(2 * pi * x[2]) * cos(2 * pi * t)
+
+    return SVector(h, h / 2, 3 * h / 2)
+end
+
+
+"""
+source_terms_eoc_test_polytropic(u, x, t, equations::PolytropicEulerEquations2D)
+
+"""
+@inline function source_terms_convergence_test(u, x, t,
```

Comment:
The initial conditions and source terms functions come before the `flux` functions within the other equation files. These should get moved for the sake of consistency.

Reply:
Done.